### ISSUE TITLE
Fails to compile with `default-features = false` and `features = ["alloc"]`

### ISSUE THREAD

**Issue by romac** [multiformats/rust-multihash#375]
`multihash` fails to compile with `default-features = false` and `features = ["alloc"]`.

Either adding the `std` feature or removing the `alloc` feature makes the build go through. Unfortunately neither option work for me because I would like to call `Multihash::to_bytes` in `no_std` + `alloc` context.

### `Cargo.toml`

```toml
[package]
name = "tmp-xqsq7b"
version = "0.1.0"
edition = "2021"

[dependencies]
multihash = { version = "0.19.2", default-features = false, features = ["alloc"] }
```

### Result

```
$ cargo check
    Checking multihash v0.19.2
error[E0277]: the trait bound `Vec<_>: core2::io::Write` is not satisfied
   --> /Users/romac/.cargo/registry/src/index.crates.io-6f17d22bba15001f/multihash-0.19.2/src/multihash.rs:141:25
    |
141 |             .write(&mut bytes)
    |              -----      ^^^^^ the trait `core2::io::Write` is not implemented for `Vec<_>`, which is required by `&mut Vec<_>: core2::io::Write`
    |              |
    |              required by a bound introduced by this call
    |
    = help: the following other types implement trait `core2::io::Write`:
              &mut W
              &mut [u8]
              core2::io::Cursor<&mut [u8]>
    = note: required for `&mut Vec<_>` to implement `core2::io::Write`
note: required by a bound in `Multihash::<S>::write`
   --> /Users/romac/.cargo/registry/src/index.crates.io-6f17d22bba15001f/multihash-0.19.2/src/multihash.rs:121:21
    |
121 |     pub fn write<W: io::Write>(&self, w: W) -> Result<usize, Error> {
    |                     ^^^^^^^^^ required by this bound in `Multihash::<S>::write`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `multihash` (lib) due to 1 previous error
```

**Comment by vmx** [comment]
I've released v0.19.3 that contains a fix. Thanks @Stebalien for the short turnaround time!

**Comment by romac** [comment]
Thank you so much for the quick turnaround! 🙏
